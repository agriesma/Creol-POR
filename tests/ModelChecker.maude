***
*** Test suite for the model checker.
***
set show timing off .
set show advise off .
load modelchecker .

quit
