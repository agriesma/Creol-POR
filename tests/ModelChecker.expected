==========================================
reduce in TEST-TAILCALL : init .
result Configuration: 
< ob('Tailcall0) : Qu | Size: 10,Dealloc: noDealloc,Ev: comp(label(ob(
    'Tailcall0), ob('Tailcall0), 'init, emp),emp) >
 
< ob('main) : Qu | Size: 1,Dealloc: noDealloc,Ev: noMsg >
 
< 'Tailcall : Cl | Inh: noInh,Par: 'x,Att: 'x |-> null,Mtds:
< 'init : Mtdname | Param: noAid,Latt: noSubst,Code: return(emp) > *
< 'run : Mtdname | Param: noAid,Latt: noSubst,Code: if 'less[[int(0) # 'x]] th
    'x ::= ('minus[['x # int(1)]]) ; tailcall 'run(emp) el return(emp) fi >,
    Ocnt: 1 >
 
< ob('Tailcall0) : 'Tailcall | Att: 'this |-> ob('Tailcall0), 'x |-> int(5),Pr:
    ('caller |-> ob('Tailcall0), 'label |-> label(ob('Tailcall0), ob(
    'Tailcall0), 'init, emp)),
cont(label(ob('Tailcall0), ob('Tailcall0), 'init, emp)),PrQ: 'Dummy |-> label(
    ob('Tailcall0), ob('Tailcall0), 'init, emp),
label(ob('Tailcall0), ob('Tailcall0), 'init, emp) ?(noAid) ; 'Dummy ! 'run(emp)
    ; 'Dummy ?(noAid),Lcnt: 1 >
 
< ob('main) : 'Class | Att: noSubst,Pr: idle,PrQ: noProc,Lcnt: 0 >

==========================================
rewrite [1] in TEST-TAILCALL : init .
result Configuration: 
< ob('Tailcall0) : Qu | Size: 10,Dealloc: noDealloc,Ev: noMsg >
 
< ob('main) : Qu | Size: 1,Dealloc: noDealloc,Ev: noMsg >
 
< 'Tailcall : Cl | Inh: noInh,Par: 'x,Att: 'x |-> null,Mtds:
< 'init : Mtdname | Param: noAid,Latt: noSubst,Code: return(emp) > *
< 'run : Mtdname | Param: noAid,Latt: noSubst,Code: if 'less[[int(0) # 'x]] th
    'x ::= ('minus[['x # int(1)]]) ; tailcall 'run(emp) el return(emp) fi >,
    Ocnt: 1 >
 
< ob('Tailcall0) : 'Tailcall | Att: 'this |-> ob('Tailcall0), 'x |-> int(4),Pr:
    noSubst,
accept(tag(label(ob('Tailcall0), ob('Tailcall0), 'run, emp))),PrQ: ('Dummy |->
    label(ob('Tailcall0), ob('Tailcall0), 'run, emp),
label(ob('Tailcall0), ob('Tailcall0), 'run, emp) ?(noAid)) ++ ('caller |-> ob(
    'Tailcall0), 'label |-> tag(label(ob('Tailcall0), ob('Tailcall0), 'run,
    emp))),
if 'less[[int(0) # 'x]] th 'x ::= ('minus[['x # int(1)]]) ; tailcall 'run(emp)
    el return(emp) fi,Lcnt: 1 >
 
< ob('main) : 'Class | Att: noSubst,Pr: idle,PrQ: noProc,Lcnt: 0 >

==========================================
rewrite in TEST-TAILCALL : init .
result Configuration: 
< ob('Tailcall0) : Qu | Size: 10,Dealloc: noDealloc,Ev: noMsg >
 
< ob('main) : Qu | Size: 1,Dealloc: noDealloc,Ev: noMsg >
 
< 'Tailcall : Cl | Inh: noInh,Par: 'x,Att: 'x |-> null,Mtds:
< 'init : Mtdname | Param: noAid,Latt: noSubst,Code: return(emp) > *
< 'run : Mtdname | Param: noAid,Latt: noSubst,Code: if 'less[[int(0) # 'x]] th
    'x ::= ('minus[['x # int(1)]]) ; tailcall 'run(emp) el return(emp) fi >,
    Ocnt: 1 >
 
< ob('Tailcall0) : 'Tailcall | Att: 'this |-> ob('Tailcall0), 'x |-> int(0),Pr:
    idle,PrQ: noProc,Lcnt: 1 >
 
< ob('main) : 'Class | Att: noSubst,Pr: idle,PrQ: noProc,Lcnt: 0 >

Bye.
